File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {

//Two Pointers Approach
ListNode* dummy=new ListNode(0,head);

ListNode* left=dummy;
ListNode* right=head;

while(n--){
    right=right->next;
}

while(right){
    left=left->next;
    right=right->next;
}
//current left is at one node before the node we want to remove

ListNode* curr=left->next;
ListNode* temp=curr->next;

left->next=temp;
curr->next=nullptr;


return dummy->next;
    }
};

File: Data Structures & Algorithms/reverse-a-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None or head.next is None:
            return head
        prev=head
        curr=head.next
        temp=curr.next

        while temp is not None:
            curr.next=prev
            if prev==head:
                prev.next=None
            prev=curr
            curr=temp
            temp=temp.next
        curr.next=prev
        if prev==head:
            prev.next=None
        return curr
        